app/assets/javascripts/discourse/app/routes/discovery-edit-category.js

```diff
@@ -0,0 +1,28 @@
+import I18n from "I18n";
+import DiscourseRoute from "discourse/routes/discourse";
+import Category from "discourse/models/category";
+
+export default DiscourseRoute.extend({
+  model(params) {
+    return Category.reloadBySlugPath(params.slug).then((result) => {
+      const record = this.store.createRecord("category", result.category);
+      record.setupGroupsAndPermissions();
```

Comment:
is any of this async ?

Reply:
I don't understand the question, can you rephrase?

Reply:
Im asking if any of the functions called inside the block are async given you dont wait for it

Reply:
Ah, thanks. No, none of the functions inside the block are async.